src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,57 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
 
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(new File(toFileName), makeReport(readLinesFromFile(fromFileName)));
+    }
+
+    private StringBuilder readLinesFromFile(String fromFileName) {
+        File fileFrom = new File(fromFileName);
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileFrom))) {
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+                value = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read the file ", e);
+        }
+        return stringBuilder;
+    }
+
+    private String[] makeReport(StringBuilder linesFromFile) {
+        int supply = 0;
+        int buy = 0;
+        String[] resultString = linesFromFile.toString().split("\\W+");
```

Comment:
regex "\\W+" covers a lot of ,characters to be splited by. It's better to use the same exact character in "readFromFile" method and here.
```suggestion
        String[] linesArray = linesFromFile.toString().split(" ");
```